docs/language/inline_assembly.rst

```diff
@@ -0,0 +1,116 @@
+Inline Assembly
+===============
+
+In Solidity functions, developers are allowed to write assembly blocks containing Yul code. For more information about
+the Yul programming language, please refer to the :ref:`yul section <yul_section>`.
+
+In an assembly block, you can access solidity local variables freely and modify them as well. Bear in mind, however,
+that strings, vectors and structs are reference types, so accessing them in Yul will return a memory address. Any
```

Comment:
I would rephrase this as "reference types like string, .. are memory addresses in yul, so manipulating them can be unsafe unless done correctly"